flang/lib/Lower/Mangler.cpp

```diff
@@ -126,17 +116,44 @@ Fortran::lower::mangle::mangleName(const Fortran::semantics::Symbol &symbol) {
             return fir::NameUniquer::doCommonBlock(symbolName);
           },
           [&](const Fortran::semantics::DerivedTypeDetails &) {
+            // This is a derived type symbol definition. Note that there are
+            // no kind parameter values associated to it. Kind parameter
+            // values are defined when declaring an entity of this derived type.
             auto modNames = moduleNames(ultimateSymbol);
             auto optHost = hostName(ultimateSymbol);
-            auto kinds = kindValues(ultimateSymbol);
             return fir::NameUniquer::doType(modNames, optHost, symbolName,
-                                            kinds);
+                                            /*kinds*/ llvm::None);
           },
           [](const auto &) -> std::string { TODO_NOLOC("symbol mangling"); },
       },
       ultimateSymbol.details());
 }
 
+std::string Fortran::lower::mangle::mangleName(
+    const Fortran::semantics::DerivedTypeSpec &derivedType) {
+  // Resolve host and module association before mangling
+  const auto &ultimateSymbol = derivedType.typeSymbol().GetUltimate();
+  auto symbolName = toStringRef(ultimateSymbol.name());
+  auto modNames = moduleNames(ultimateSymbol);
+  auto optHost = hostName(ultimateSymbol);
+  std::vector<std::int64_t> kinds;
```

Comment:
Can use llvm::SmallVector since the loop was manually inlined.

Reply:
changed to SmallVector